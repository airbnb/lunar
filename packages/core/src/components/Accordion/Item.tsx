import React from 'react';
import withStyles, { WithStylesProps } from '../../composers/withStyles';
import ExpandableIcon from '../ExpandableIcon';

export type Props = {
  /** Apply a border. */
  bordered?: boolean;
  /** Content to render if the accordion item is expanded. */
  children?: React.ReactNode;
  /** Whether the accordion item is expanded or not. */
  expanded?: boolean;
  /** Unique id of the accordion item. */
  id: string;
  /** Index amongst a collection of accordion items. */
  index?: number;
  /** Title of the accordion item. */
  title?: React.ReactNode;
  /** Callback fired when the accordion item is clicked. */
  onClick?: (index: number) => void;
};

/**
 * A single accordion item. To be rendered amongst a collection of accordion items, within an accordion.
 */
export class AccordionItem extends React.Component<Props & WithStylesProps> {
  static defaultProps = {
    bordered: false,
    children: null,
    expanded: false,
  };

  private handleClick = () => {
    if (this.props.onClick) {
      this.props.onClick(this.props.index!);
    }
  };

  render() {
    const { cx, bordered, children, expanded, id, styles, theme, title } = this.props;

    return (
      <div className={cx(styles.item, bordered && styles.item_bordered)}>
        <button
          className={cx(styles.title)}
          aria-controls={`accordion-body-${id}`}
          aria-selected={expanded}
          id={`accordion-title-${id}`}
          onClick={this.handleClick}
          role="tab"
          tabIndex={0}
          type="button"
        >
          {title && <span className={cx(styles.titleText)}>{title}</span>}

          <ExpandableIcon expanded={!!expanded} size={theme!.unit * 3} />
        </button>

        <section
          className={cx(styles.body, expanded && styles.body_expanded)}
          aria-hidden={!expanded}
          aria-labelledby={`accordion-title-${id}`}
          id={`accordion-body-${id}`}
          role="tabpanel"
        >
          {children}
        </section>
      </div>
    );
  }
}

export default withStyles(
  ({ color, pattern, ui, unit }) => ({
    body: {
      display: 'none',
      padding: `${unit}px ${unit * 2}px ${unit * 2}px`,
    },

    body_expanded: {
      display: 'block',
    },

    item: {
      background: color.accent.bg,
    },

    item_bordered: {
      borderTop: ui.border,
    },

    title: {
      ...pattern.resetButton,
      display: 'flex',
      alignItems: 'center',
      padding: unit * 2,
      textAlign: 'left',
      width: '100%',
    },

    titleText: {
      flex: '1',
    },
  }),
  {
    passThemeProp: true,
  },
)(AccordionItem);
