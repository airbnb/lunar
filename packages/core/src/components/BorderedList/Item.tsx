import React from 'react';
import { mutuallyExclusiveTrueProps } from 'airbnb-prop-types';
import withStyles, { css, WithStylesProps } from '../../composers/withStyles';

const paddingPropType = mutuallyExclusiveTrueProps('compact', 'spacious');

export type Props = {
  /** Item content. */
  children: NonNullable<React.ReactNode>;
  /** Render with reduced vertical padding. */
  compact?: boolean;
  /** Render with vertical padding. */
  spacious?: boolean;
};

export class BorderedListItem extends React.Component<Props & WithStylesProps> {
  static propTypes = {
    compact: paddingPropType,
    spacious: paddingPropType,
  };

  static defaultProps = {
    compact: false,
    spacious: false,
  };

  render() {
    const { children, compact, spacious, styles } = this.props;

    return (
      <li {...css(styles.item, compact && styles.item_compact, spacious && styles.item_spacious)}>
        {children}
      </li>
    );
  }
}

export default withStyles(({ ui, unit }) => ({
  item: {
    borderTop: ui.border,

    ':last-child': {
      borderBottom: ui.border,
    },
  },

  item_compact: {
    paddingBottom: unit * 1.5,
    paddingTop: unit * 1.5,
  },

  item_spacious: {
    paddingBottom: unit * 3,
    paddingTop: unit * 3,
  },
}))(BorderedListItem);
