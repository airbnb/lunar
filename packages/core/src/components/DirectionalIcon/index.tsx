import React, { useContext } from 'react';
import { DirectionContext } from 'aesthetic-react';
import { WithIconWrapperProps } from '@airbnb/lunar-icons/lib/withIcon';
import Core from '../..';

export type Props = {
  /** The default direction to render. */
  direction: 'left' | 'right';
  /** The left direction facing icon. */
  left: React.ComponentType<WithIconWrapperProps>;
  /** The right direction facing icon. */
  right: React.ComponentType<WithIconWrapperProps>;
};

export default function DirectionalIcon({
  direction,
  left: LeftIcon,
  right: RightIcon,
  ...props
}: Props & WithIconWrapperProps) {
  const context = useContext(DirectionContext);
  const rtl = Core.isRTL(context);

  if (direction === 'left') {
    return rtl ? <RightIcon {...props} /> : <LeftIcon {...props} />;
  }

  return rtl ? <LeftIcon {...props} /> : <RightIcon {...props} />;
}
