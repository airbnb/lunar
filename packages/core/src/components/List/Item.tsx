import React from 'react';
import { mutuallyExclusiveTrueProps } from 'airbnb-prop-types';
import withStyles, { WithStylesProps } from '../../composers/withStyles';

const paddingPropType = mutuallyExclusiveTrueProps('compact', 'spacious');

export type Props = {
  /** Render with a top/bottom borders. Last item will have both. */
  bordered?: boolean;
  /** Item content. */
  children: NonNullable<React.ReactNode>;
  /** Render with reduced vertical padding. */
  compact?: boolean;
  /** @ignore */
  horizontal?: boolean;
  /** Render with vertical padding. */
  spacious?: boolean;
};

export class ListItem extends React.Component<Props & WithStylesProps> {
  static propTypes = {
    compact: paddingPropType,
    spacious: paddingPropType,
  };

  static defaultProps = {
    bordered: false,
    compact: false,
    horizontal: false,
    spacious: false,
  };

  render() {
    const { bordered, children, compact, cx, horizontal, spacious, styles } = this.props;

    return (
      <li
        className={cx(
          !horizontal && bordered && styles.item_bordered,
          horizontal && bordered && styles.item_bordered_horizontal,
          !horizontal && compact && styles.item_compact,
          horizontal && compact && styles.item_compact_horizontal,
          !horizontal && spacious && styles.item_spacious,
          horizontal && spacious && styles.item_spacious_horizontal,
        )}
      >
        {children}
      </li>
    );
  }
}

export default withStyles(({ ui, unit }) => ({
  item_bordered: {
    borderTop: ui.border,

    ':last-child': {
      borderBottom: ui.border,
    },
  },

  item_bordered_horizontal: {
    borderLeft: ui.border,

    ':last-child': {
      borderRight: ui.border,
    },
  },

  item_compact: {
    paddingBottom: unit * 1.5,
    paddingTop: unit * 1.5,
  },

  item_compact_horizontal: {
    paddingLeft: unit * 1.5,
    paddingRight: unit * 1.5,
  },

  item_spacious: {
    paddingBottom: unit * 3,
    paddingTop: unit * 3,
  },

  item_spacious_horizontal: {
    paddingLeft: unit * 3,
    paddingRight: unit * 3,
  },
}))(ListItem);
