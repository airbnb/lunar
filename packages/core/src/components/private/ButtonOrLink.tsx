import React from 'react';
import removeFocusOnMouseUp from '../../utils/removeFocusOnMouseUp';
import IconAffix from './IconAffix';

export type ButtonOrLinkTypes = HTMLAnchorElement | HTMLButtonElement;

export type Props = {
  [prop: string]: any;
  /** Icon to display at the end the content. */
  afterIcon?: React.ReactNode;
  /** Icon to display at the start the content. */
  beforeIcon?: React.ReactNode;
  /** Content within the button or link. */
  children: NonNullable<React.ReactNode>;
  /** Whether the element is disabled. */
  disabled?: boolean;
  /** Apply flexbox styles to icons. */
  flexAlign?: boolean;
  /** Render as an anchor link with a URL. */
  href?: string;
  /** Whether the element is loading. */
  loading?: boolean;
  /** Callback fired when the element is clicked. */
  onClick?: (event: React.MouseEvent<ButtonOrLinkTypes>) => void;
  /** Callback fired when the element is released. */
  onMouseUp?: (event: React.MouseEvent<ButtonOrLinkTypes>) => void;
  /** When a link, open the target in a new window. */
  openInNewWindow?: boolean;
  /** Rel attribute override for if the component has an href */
  rel?: string;
  /** When a button, the type of button. */
  type?: 'button' | 'submit' | 'reset';
};

export default class ButtonOrLink extends React.Component<Props> {
  static defaultProps = {
    afterIcon: null,
    beforeIcon: null,
    disabled: false,
    flexAlign: false,
    href: '',
    loading: false,
    openInNewWindow: false,
    type: 'button',
  };

  // Intercept click to handle disabled state
  private handleClick = (event: React.MouseEvent<ButtonOrLinkTypes>) => {
    if (this.props.disabled) {
      event.preventDefault();
    } else if (this.props.onClick) {
      this.props.onClick(event);
    }
  };

  // Remove outline after click
  private handleMouseUp = (event: React.MouseEvent<ButtonOrLinkTypes>) => {
    removeFocusOnMouseUp(event);

    if (this.props.onMouseUp) {
      this.props.onMouseUp(event);
    }
  };

  render() {
    const {
      afterIcon,
      beforeIcon,
      children,
      disabled,
      flexAlign,
      href,
      loading,
      openInNewWindow,
      rel,
      type,
      ...restProps
    } = this.props;
    const Tag = href ? 'a' : 'button';
    const props: any = {};

    // Determine props based on element type
    if (href) {
      props.href = href;
      props.rel = rel;

      if (openInNewWindow) {
        props.target = '_blank';

        if (props.rel === undefined) {
          props.rel = 'noopener noreferrer';
        }
      }
    } else {
      props.disabled = disabled || loading || false;
      props.type = type || 'button';
    }

    return (
      // @ts-ignore [ts] JSX element type 'Component' does not have any construct or call signatures. [2604]
      <Tag {...restProps} {...props} onClick={this.handleClick} onMouseUp={this.handleMouseUp}>
        {!loading && beforeIcon && (
          <IconAffix before flex={flexAlign}>
            {beforeIcon}
          </IconAffix>
        )}

        <span>{children}</span>

        {!loading && afterIcon && (
          <IconAffix after flex={flexAlign}>
            {afterIcon}
          </IconAffix>
        )}
      </Tag>
    );
  }
}
