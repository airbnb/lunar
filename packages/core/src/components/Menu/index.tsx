import React from 'react';
import withStyles, { WithStylesProps } from '../../composers/withStyles';
import Separator from './Separator';
import Item from './Item';
import Row from './Row';

export type Props = {
  /** List of `Row`s, `Item`s, and `Separator`s to render in the menu. */
  children?: React.ReactNode;
  /** Accessibility label. */
  accessibilityLabel: string;
  /** Maximum height of the menu before scrolling. */
  maxHeight?: number;
  /** Minimum width of the menu. */
  minWidth?: number;
  /** Accessibility role. */
  role?: string;
  /** Whether or not the menu has visible overflow. */
  overflow?: boolean;
};

/** An abstract menu for use within dropdowns, selects, autocompletes, and more. */
export class Menu extends React.Component<Props & WithStylesProps> {
  static defaultProps = {
    children: null,
    minWidth: 200,
    overflow: false,
    role: 'menu',
  };

  render() {
    const {
      cx,
      accessibilityLabel,
      children,
      maxHeight,
      minWidth,
      overflow,
      role,
      styles,
    } = this.props;
    const scrollable = !!maxHeight && !overflow;

    return (
      <ul
        role={role}
        aria-label={accessibilityLabel}
        className={cx(
          styles.menu,
          { maxHeight: scrollable ? maxHeight : 'auto', minWidth },
          scrollable && styles.menu_scrollable,
        )}
      >
        {children}
      </ul>
    );
  }
}

export { Item, Separator, Row };

export default withStyles(({ color, ui, pattern }) => ({
  menu: {
    ...pattern.box,
    margin: 0,
    padding: 0,
    backgroundColor: color.accent.bg,
    listStyle: 'none',

    '@selectors': {
      '> li': {
        position: 'relative',
      },

      // These are jank. Better way?
      '> li:first-child > *': {
        borderTopLeftRadius: ui.borderRadius,
        borderTopRightRadius: ui.borderRadius,
      },

      '> li:last-child > *': {
        borderBottomLeftRadius: ui.borderRadius,
        borderBottomRightRadius: ui.borderRadius,
      },
    },
  },

  menu_scrollable: {
    overflowY: 'auto',
  },
}))(Menu);
