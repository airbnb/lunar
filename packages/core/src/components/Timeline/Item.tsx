import React from 'react';
import IconRecord from '@airbnb/lunar-icons/src/interface/IconRecord';
import useStyles from '../../hooks/useStyles';
import { styleSheetItem } from './styles';
import Text from '../Text';
import useTheme from '../../hooks/useTheme';
import DateTime from '../DateTime';
import { DateTimeType } from '../../types';
import T from '../Translate';
import { Item as ListItem } from '../List';

export type ItemProps = {
  /** Datetime occurence of the item. Not used for secondary items. */
  at?: DateTimeType;
  /** Whether an item is the oldest (chronologically) in the collection of items. */
  oldest?: boolean;
  /** Whether an item should be visually deemphasized. */
  secondary?: boolean;
  /** Content to render for an item. */
  children: NonNullable<React.ReactNode>;
};

/** Represents a single item in a collection of chronologically items. */
export default function Item({ at, oldest = false, secondary = false, children }: ItemProps) {
  const theme = useTheme();
  const [styles, cx] = useStyles(styleSheetItem);

  return (
    <ListItem>
      <div className={cx(styles.wrapper)}>
        <div className={cx(styles.barWrapper)}>
          <div className={cx(styles.iconWrapper)}>
            <IconRecord
              decorative
              size={secondary ? theme.unit * 0.5 : theme.unit}
              color={theme.color.core.neutral[3]}
            />
          </div>

          {!oldest && <div className={cx(styles.bar)} />}
        </div>

        <div
          className={cx(
            styles.content,
            secondary && styles.content_secondary,
            oldest && styles.content_oldest,
          )}
        >
          {!secondary && (
            <Text small muted>
              <T
                phrase="%{mediumDate} (%{relativeDate})"
                k="lunar.timeline.date"
                mediumDate={DateTime.format({ at, medium: true })}
                relativeDate={DateTime.relative(at!)}
              />
            </Text>
          )}

          {children}
        </div>
      </div>
    </ListItem>
  );
}
