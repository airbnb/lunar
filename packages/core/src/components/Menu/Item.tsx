import React from 'react';
import IconCaretLeft from '@airbnb/lunar-icons/lib/interface/IconCaretLeft';
import IconCaretRight from '@airbnb/lunar-icons/lib/interface/IconCaretRight';
import iconComponent from '../../prop-types/iconComponent';
import withStyles, { WithStylesProps } from '../../composers/withStyles';
import ButtonOrLink from '../private/ButtonOrLink';
import Text from '../Text';
import DirectionalIcon from '../DirectionalIcon';

export type Props = {
  /** Content within the menu item. */
  children: NonNullable<React.ReactNode>;
  /** Mark the button as disabled. */
  disabled?: boolean;
  /** Mark the item as highlighted. */
  highlighted?: boolean;
  /** Render an anchor link with a URL instead of a button. */
  href?: string;
  /** An icon to display before the item. */
  icon?: React.ReactNode;
  /** Click handler. */
  onClick?: () => void;
  /** Opens links in a new window. */
  openInNewWindow?: boolean;
  /** Accessibility role. */
  role?: string;
  /** Increase padding. */
  spacious?: boolean;
  /** A sub-menu to display on hover. */
  submenu?: React.ReactNode;
  /** Tab index for the current menu. */
  tabIndex?: number;
  /** Tip to display after the item. */
  tip?: React.ReactNode;
};

/** An interactive item within a menu. */
export class MenuItem extends React.Component<Props & WithStylesProps> {
  static propTypes = {
    icon: iconComponent,
  };

  static defaultProps = {
    disabled: false,
    highlighted: false,
    href: '',
    icon: null,
    openInNewWindow: false,
    role: 'menuitem',
    spacious: false,
    submenu: null,
    tabIndex: -1,
    tip: null,
  };

  state = {
    showSubmenu: false,
  };

  private handleMouseEnter = () => {
    if (this.props.submenu) {
      this.setState({
        showSubmenu: true,
      });
    }
  };

  private handleMouseLeave = () => {
    if (this.props.submenu) {
      this.setState({
        showSubmenu: false,
      });
    }
  };

  render() {
    const {
      cx,
      children,
      disabled,
      highlighted,
      href,
      icon,
      onClick,
      openInNewWindow,
      role,
      spacious,
      styles,
      submenu,
      tabIndex,
      tip,
    } = this.props;
    const { showSubmenu } = this.state;
    const after = submenu ? (
      <DirectionalIcon
        direction="right"
        left={IconCaretLeft}
        right={IconCaretRight}
        size="1.5em"
        decorative
      />
    ) : (
      tip
    );

    return (
      <li role="none" onMouseEnter={this.handleMouseEnter} onMouseLeave={this.handleMouseLeave}>
        <ButtonOrLink
          afterIcon={
            after ? (
              <Text muted small>
                {after}
              </Text>
            ) : null
          }
          beforeIcon={icon}
          disabled={disabled}
          href={href}
          onClick={onClick}
          openInNewWindow={openInNewWindow}
          role={role}
          tabIndex={tabIndex}
          aria-haspopup={!!submenu}
          aria-expanded={showSubmenu}
          className={cx(
            styles.item,
            (showSubmenu || highlighted) && styles.item_highlighted,
            disabled && styles.item_disabled,
            spacious && styles.item_spacious,
          )}
        >
          {children}
        </ButtonOrLink>

        {showSubmenu && <div className={cx(styles.submenu)}>{submenu}</div>}
      </li>
    );
  }
}

export default withStyles(({ color, font, pattern, unit, transition }) => ({
  item: {
    ...transition.box,
    ...font.textRegular,
    width: '100%',
    padding: `${unit}px ${1.5 * unit}px`,
    border: 0,
    textAlign: 'left',
    backgroundColor: 'transparent',
    cursor: 'pointer',
    textDecoration: 'none',
    outlineOffset: '-3px',
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    color: color.accent.text,

    ':hover': {
      backgroundColor: color.accent.bgHover,
      color: color.core.neutral[6],
    },

    '@selectors': {
      // Fix content
      '> span': {
        flexGrow: 1,
      },

      // Fix icons
      '> div': {
        flexGrow: 0,
        margin: 0,

        ':first-child': { marginRight: unit },
        ':last-child': { marginLeft: unit },
      },
    },
  },

  item_spacious: {
    padding: unit * 2,
  },

  item_disabled: {
    ...pattern.disabled,
  },

  item_highlighted: {
    backgroundColor: color.accent.bgHover,
  },

  submenu: {
    position: 'absolute',
    top: -1,
    left: '99%',
  },
}))(MenuItem);
