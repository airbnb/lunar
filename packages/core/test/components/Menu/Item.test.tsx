import React from 'react';
import { shallowWithStyles } from '@airbnb/lunar-test-utils';
import IconCaretRight from '@airbnb/lunar-icons/lib/interface/IconCaretRight';
import Menu from '../../../src/components/Menu';
import Item from '../../../src/components/Menu/Item';
import ButtonOrLink from '../../../src/components/private/ButtonOrLink';
import DirectionalIcon from '../../../src/components/DirectionalIcon';

describe('<MenuItem />', () => {
  it('renders highlighted', () => {
    const wrapper = shallowWithStyles(<Item highlighted>Foo</Item>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders spacious', () => {
    const wrapper = shallowWithStyles(<Item spacious>Foo</Item>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a list with the correct role', () => {
    const wrapper = shallowWithStyles(<Item>Foo</Item>);

    expect(wrapper.is('li')).toBe(true);
    expect(wrapper.prop('role')).toBe('none');
  });

  it('adds role to button', () => {
    const wrapper = shallowWithStyles(<Item>Foo</Item>);

    expect(wrapper.find(ButtonOrLink).prop('role')).toBe('menuitem');

    wrapper.setProps({
      role: 'option',
    });

    expect(wrapper.find(ButtonOrLink).prop('role')).toBe('option');
  });

  it('passes props to underlying button', () => {
    const wrapper = shallowWithStyles(
      <Item disabled openInNewWindow href="/" tabIndex={0}>
        Foo
      </Item>,
    );

    expect(wrapper.find(ButtonOrLink).prop('disabled')).toBe(true);
  });

  it('passes icon to underlying button', () => {
    const icon = <IconCaretRight decorative />;
    const wrapper = shallowWithStyles(<Item icon={icon}>Foo</Item>);

    expect(wrapper.find(ButtonOrLink).prop('beforeIcon')).toBe(icon);
  });

  it('passes tip to underlying button', () => {
    const wrapper = shallowWithStyles(<Item>Foo</Item>);

    expect(wrapper.find(ButtonOrLink).prop('afterIcon')).toBeNull();

    wrapper.setProps({
      tip: 'Something',
    });

    const icon = wrapper.find(ButtonOrLink).prop('afterIcon');
    const iconWrapper = shallowWithStyles(icon as any, true);

    expect(icon).not.toBeNull();
    expect(iconWrapper.prop('children')).toBe('Something');
  });

  describe('submenu', () => {
    const menu = (
      <Menu accessibilityLabel="Test">
        <Item>Sub</Item>
      </Menu>
    );

    it('displays a submenu when item is hovered', () => {
      const wrapper = shallowWithStyles(<Item submenu={menu}>Parent</Item>);

      expect(wrapper.contains(menu)).toBe(false);

      wrapper.find('li').simulate('mouseEnter');

      expect(wrapper.contains(menu)).toBe(true);

      wrapper.find('li').simulate('mouseLeave');

      expect(wrapper.contains(menu)).toBe(false);
    });

    it('sets aria-haspopup', () => {
      const wrapper = shallowWithStyles(<Item submenu={menu}>Parent</Item>);

      expect(wrapper.find(ButtonOrLink).prop('aria-haspopup')).toBe(true);

      wrapper.setProps({
        submenu: null,
      });

      expect(wrapper.find(ButtonOrLink).prop('aria-haspopup')).toBe(false);
    });

    it('sets aria-expanded', () => {
      const wrapper = shallowWithStyles(<Item submenu={menu}>Parent</Item>);

      expect(wrapper.find(ButtonOrLink).prop('aria-expanded')).toBe(false);

      wrapper.find('li').simulate('mouseEnter');

      expect(wrapper.find(ButtonOrLink).prop('aria-expanded')).toBe(true);
    });

    it('sets after icon when a submenu exists', () => {
      const wrapper = shallowWithStyles(<Item>Foo</Item>);

      expect(wrapper.find(ButtonOrLink).prop('afterIcon')).toBeNull();

      wrapper.setProps({
        submenu: menu,
      });

      const icon = wrapper.find(ButtonOrLink).prop('afterIcon');
      const iconWrapper = shallowWithStyles(icon as any, true);

      expect(icon).not.toBeNull();
      expect(iconWrapper.find(DirectionalIcon)).toHaveLength(1);
    });
  });
});
