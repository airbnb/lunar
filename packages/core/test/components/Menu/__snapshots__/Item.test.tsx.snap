// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> renders highlighted 1`] = `
<li
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="none"
>
  <ButtonOrLink
    aria-expanded={false}
    aria-haspopup={false}
    className="item_12t53i1-o_O-item_highlighted_16ovkzb"
    role="menuitem"
    tabIndex={-1}
  >
    Foo
  </ButtonOrLink>
</li>
`;

exports[`<MenuItem /> renders spacious 1`] = `
<li
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="none"
>
  <ButtonOrLink
    aria-expanded={false}
    aria-haspopup={false}
    className="item_12t53i1-o_O-item_spacious_1x0fg6n"
    role="menuitem"
    tabIndex={-1}
  >
    Foo
  </ButtonOrLink>
</li>
`;
